DOC: Update broadcasting doc with current exception details

tgsmith61591 opened the pull request:
The exception details in the broadcast documentation were old and have since been updated. This commit changes the documentation to reflect the current exception details when a shape mismatch is raised.

Old:

> If these conditions are not met, a `ValueError: frames are not aligned` exception is thrown

New:

> If these conditions are not met, a `ValueError: operands could not be broadcast together` exception is thrown

There was a similar issue in an example in the documentation:

Old:

```python
 >>> x + y
 <type 'exceptions.ValueError'>: shape mismatch: objects cannot be broadcast to a single shape
```

New:

```python
 >>> x + y
 <type 'exceptions.ValueError'>: operands could not be broadcast together with shapes (4,) (5,)
```

<!-- Please be sure you are following the instructions in the dev guidelines
http://www.numpy.org/devdocs/dev/gitwash/development_workflow.html
-->

<!-- We'd appreciate it if your commit message is properly formatted
http://www.numpy.org/devdocs/dev/gitwash/development_workflow.html#writing-the-commit-message
-->

tgsmith61591 committed:
DOC: Update broadcasting documentation

The exception details in the broadcast documentation were old and
have since been updated. This commit changes the documentation to
reflect the current exception details when a shape mismatch is
raised.

eric-wieser commented on numpy/doc/broadcasting.py:
```diff
@@ -124,7 +125,7 @@
 
  >>> x + y
- <type 'exceptions.ValueError'>: shape mismatch: objects cannot be broadcast to a single shape
+ <type 'exceptions.ValueError'>: operands could not be broadcast together with shapes (4,) (5,)
```
Can you replace `<type 'exceptions.ValueError'>` with `ValueError`?

tgsmith61591 replied:
Done!

tgsmith61591 committed:
DOC: Replace <type 'exceptions.ValueError'> with ValueError

Replaced <type 'exceptions.ValueError'> with ValueError in the
outdated broadcasting documentation

eric-wieser commented on numpy/doc/broadcasting.py:
```diff
@@ -53,9 +53,10 @@
+``ValueError: operands could not be broadcast together`` exception is 
+thrown, indicating that the arrays have incompatible shapes. The size of 
+the resulting array is the maximum size along each dimension of the input 
+arrays.
```
nit: could replace `size` with `shape` in this sentence, but not really related to this patch, so feel free to leave as is

mattip commented:
Thanks @tgsmith61591
